### ISSUE TITLE
Visual Studio 17.12 and .NET 8 requires XCode 16

### ISSUE THREAD

**Issue by eqmarcin** [dotnet/maui#25888]
### Description

I have updated Visual Studio Community to from 17.11.4 to 17.12.0 and now I can't build/run iOS apps on paired Mac (Mac OS Sonoma 14.7.1 with XCode 15.4) as it says that XCode 16 is required.

I'm using .NET 8.0.404 and workloads below (this setup works on another PC with VS 17.11.4)
![Image](https://github.com/user-attachments/assets/4ec1981c-1700-4ae4-8e0b-18d7a023b136)

Errors:
`This version of Microsoft.iOS requires the iOS 18.0 SDK (shipped with Xcode 16.0). Either upgrade Xcode to get the required header files or set the managed linker behaviour to Link Framework SDKs Only in your project's iOS Build Options > Linker Behavior (to try to avoid the new APIs).`
`The linker step 'Setup' failed during processing: This version of Microsoft.iOS requires the iOS 18.0 SDK (shipped with Xcode 16.0). Either upgrade Xcode to get the required header files or set the managed linker behaviour to Link Framework SDKs Only in your project's iOS Build Options > Linker Behavior (to try to avoid the new APIs).`

Changing linker setting to "Link Framework SDKs Only" fixes the above for builds but it disables Hot Reload ("iOS Linker settings for MyApp are unsupported").

Not sure if there is anything that I can do to have it working as before?

Notes:
I can't downgrade VS version this feature is not available for Community edition.
Upgrading Mac OS and XCode to v16 and/or .NET to 9 breaks the app.

### Version with bug

8.0.80 SR8

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, I was *not* able test on other platforms

### Relevant log output

```shell

```

**Comment by brianmathews** [comment]
You are not alone.  I have the same issue.

**Comment by brianmathews** [comment]
Oh wow, I didn't expect a response! Cool!

I **am** using Xcode 16, and it was working including submitting to the Apple App Store.  The problems started when I upgraded to **VS 17.12**, and when I upgraded some NuGet packages.  I use VS on Windows primarily, and use Pair-to-Mac when making builds for the Apple App Store.  

So I don't have the exact same issue as the original poster, but I get the same error messages and duplicative versions of MAUI workloads.

I had everything working on VS 17.11 (or whatever version came before 17.12.0):
- Xcode 16, 
- .Net 8.0.40x (I think it was 8.0.402?  You posted an article when Xcode 16 first came out to use MAUI 8.0.402.1 and I followed it).

After installing VS 17.12 it made breaking changes:
- It installed .Net 9 and started using 9 though I didn't want to upgrade (`dotnet --version` reported I was now using 9)
- It installed multiple copies of the .Net MAUI workloads (`dotnet workload list` showed both 8.0.400 and 17.12.xxxx versions for all the MAUI workloads)
- New compiler errors about Android ABI stuff in my .csproj
    - Had to change from: `<AndroidSupportedAbis>armeabi-v7a;arm64-v8a;x86;x86_64</AndroidSupportedAbis>` to: `<RuntimeIdentifiers>android-arm;android-arm64;android-x86;android-x64</RuntimeIdentifiers>`
- I started getting the error `This version of Microsoft.iOS requires the iOS 18.0 SDK (shipped with Xcode 16.0)` even though I have Xcode 16.0 on the mac
- It started complaining about the `info.plist` not being found.  After doing a few more cleanings, the issue magically went away. 
- etc.

So then I tried to get back to the previously known-working state with these steps https://github.com/xamarin/xamarin-macios/releases/tag/dotnet-8.0.1xx-xcode16.0-8303 below:
- I rolled-back my .csproj file to use the original known-working NuGet packages
- I added a `global.json` file with: `{"sdk": {"version": "8.0.402", "rollForward": "patch", "allowPrerelease": false }}`
- I did:
    - Uninstalled then Installed .Net SDK 8.0.402
    - `dotnet workload uninstall maui`
    - `dotnet workload uninstall android ios maccatalyst maui-windows`
    - `dotnet nuget locals all --clear`
    - rmdir all my `bin` and `obj` directories
    - `dotnet workload install maui --version 8.0.402.1`
    - `dotnet restore --force`

And now I have things as follows:

```
> dotnet --version
8.0.402
```
and:
```
> dotnet workload list

Workload version: 8.0.402.1

Installed Workload Id      Installation Source
----------------------------------------------------------
android                    SDK 8.0.400, VS 17.12.35506.116
aspire                     SDK 8.0.400, VS 17.12.35506.116
ios                        SDK 8.0.400, VS 17.12.35506.116
maccatalyst                SDK 8.0.400, VS 17.12.35506.116
macos                      SDK 8.0.400
maui                       SDK 8.0.400
maui-windows               SDK 8.0.400, VS 17.12.35506.116
tvos                       SDK 8.0.400
```

I tried to add my `.csproj` file for context, but this forum doesn't allow attaching .csproj files.  If you want it I can rename it to *.txt and try attaching.

### Resulting New Errors
I've gotten it running again on Windows VS when I **don't** use Pair-to-Mac.

I am getting strange and new errors with Pair-to-Mac from Windows VS:
```
Root assembly 'SQLitePCLRaw.provider.dynamic_cdecl' could not be found.
The file "obj\Release\net8.0-ios\ios-arm64\linked\Link.semaphore" cannot be created. Could not find a part of the path 'C:\swd\sotamat\SotamatApp\SotamatApp\obj\Release\net8.0-ios\ios-arm64\linked\Link.semaphore'.
```

Ideas of what to try next?

**Comment by PureWeen** [comment]
@brianmathews it'd be useful to breakup your blockers here a bit to separate issues so we can route them.

It sounds like you have two categories of issues here

1) You're blocked from updating to .NET9 because of some blockers/regressions
2) You're having troubles reverting back to .NET 8 workloads (state) and using pair to mac

If you could log issues why you're blocked from using .NET9 that would be helpful so we can fix any regressions.

And then any pair to mac issues you're hitting if you could log these through VS Feedback so we can check the relevant logs
